src/code.cloudfoundry.org/gorouter/route/pool.go

```diff
@@ -433,16 +465,29 @@ func (p *EndpointPool) removeEndpoint(e *endpointElem) {
 	delete(p.index, e.endpoint.CanonicalAddr())
 	delete(p.index, e.endpoint.PrivateInstanceId)
 	p.Update()
+
+	if p.LoadBalancingAlgorithm == config.LOAD_BALANCE_HB {
+		p.HashLookupTable.Remove(e.endpoint.PrivateInstanceId)
+	}
+
 }
 
-func (p *EndpointPool) Endpoints(logger *slog.Logger, initial string, mustBeSticky bool, azPreference string, az string) EndpointIterator {
+func (p *EndpointPool) Endpoints(logger *slog.Logger, initial string, mustBeSticky bool, azPreference string, az string, globalLB string, request *http.Request) EndpointIterator {
```

Comment:
you're passing in the full request to then dynamically fetch the hash based header. Why not pass an optional reference to `*HashRoutingProperties`? Parsing then can be done at the call site where you have the request anyway.